models/users.js

```diff
@@ -576,6 +576,52 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+const fetchUsersWithToken = async () => {
+  try {
+    const users = [];
+    const usersRef = await userModel.where("tokens", "!=", false).get();
+    usersRef.forEach((user) => {
+      users.push(userModel.doc(user.id));
+    });
+    return users;
+  } catch (err) {
+    logger.error(`Error while fetching all users with tokens field: ${err}`);
+    throw err;
+  }
+};
+
+const removeGitHubToken = async (users) => {
+  try {
+    const length = users.length;
+
+    let numberOfBatches = length / 500;
+    numberOfBatches = numberOfBatches + 1;
```

Comment:
what if there are exactly multiple of 500 users found

Reply:
Hmmm true. I ll need to check for remainder.
If there is remainder only then create one extra otherwise dont.